packages/backend/src/apps/formsg/triggers/new-submission/get-data-out-metadata.ts

```diff
@@ -50,6 +51,42 @@ function buildAnswerMetadatum(fieldData: IJSONObject): IDataOutMetadatum {
   return answer
 }
 
+function buildAnswerArrayMetadatum(
+  fieldData: IJSONObject,
+): IDataOutMetadatum[] {
+  // there should only be checkbox and table fieldtypes that contain answer array
+  const answerArray = [] as IDataOutMetadata[]
+  const array = fieldData.answerArray as IJSONArray
+  for (let i = 0; i < array.length; i++) {
+    // check for table fieldType to do more processing
+    const option = array[i]
+    if (fieldData.fieldType === 'table') {
+      const nestedAnswerArray = [] as IDataOutMetadatum[]
+      const optionArray = option as IJSONArray
+      for (let j = 0; j < optionArray.length; j++) {
+        nestedAnswerArray.push({
+          type: 'text',
+          label: fieldData.order
+            ? `Response ${fieldData.order}, Row ${i + 1} Column ${j + 1}`
+            : null,
+          order: fieldData.order ? (fieldData.order as number) : null,
+        })
+      }
+      answerArray.push(nestedAnswerArray)
+    } else {
```

Comment:
To be more defensive, I'd rather we use a `switch` and fallback to returning nothing by default.

What if FormSG introduces a new field type with answerArray?